toptal/picasso
# feat: add Button.Action

[FX-1139](https://toptal-core.atlassian.net/browse/FX-1139)

### Description

[🖼️ Abstract link](https://app.abstract.com/projects/1b06c884-06af-482a-bf12-a82f521a19a1/branches/master/commits/e1516305cb26595439fbe2ef35acb5ab2e175d1a/files/f58fa9b1-6361-4d06-a6c7-d57f8c9d715d/layers/4DA6F551-1F6D-425D-B4B0-2954C6594ADB?mode=design&sha=e1516305cb26595439fbe2ef35acb5ab2e175d1a)

**Acceptance criteria**

- Create a new child component `Button.Action` that wraps Button
- Only rectangular, it can’t be circular
- Only one size, the dimensions must not be configurable
- The button label (text) should be:

```
font-size: 14px
line-height: 22px
```

- Use **Button.Action** in the Accordion component instead of the current solution https://github.com/toptal/picasso/blob/v5/packages/picasso/src/Accordion/Accordion.tsx#L144

**Dev note**

- You can use https://github.com/toptal/picasso/blob/205f2837fbb9ba789cf7d9d68b29f956ebbdaba7/packages/picasso/src/ButtonCircular/ButtonCircular.tsx as an example to base `Button.Action` on

### How to test

- please check examples in https://picasso.toptal.net/FX-1139-create-button-action-component/?path=/story/components-folder--button

### Screenshots

![buttons](https://user-images.githubusercontent.com/1390758/94812630-90890080-0429-11eb-99bc-57f10227bba4.gif)

### Review


<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run  whole pipeline
- `@toptal-bot run danger` - Danger checks
- `@toptal-bot run lint` - Run linter
- `@toptal-bot run test` - Run jest
- `@toptal-bot run build` - Check build
- `@toptal-bot run test:visual` or `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation


</details>


[FX-1139]: https://toptal-core.atlassian.net/browse/FX-1139
